app/models/user.rb

```diff
@@ -1,2 +1,12 @@
 class User < ApplicationRecord
-end
+  
+  belongs_to :category
+  has_many :started_test
+
+  def users_history_by_level (level)
+    Test.joins(:started_test)
+      .where(started_test: {user_id: self.id})
+      .where('level = ?',level)     
+      .pluck(:title)
+  end
```

Comment:
```suggestion
def users_history_by_level(level)
    Test.joins('JOIN started_tests ON started_tests.test_id = tests.id')
        .where(started_tests: { user_id: id }, tests: { level: level })
  end
```
давай как-то так решим? можешь свой вариант накинуть, но без методов ассоциаций, тк это некст тема)